azurerm/resource_arm_private_link_endpoint.go

```diff
@@ -0,0 +1,356 @@
+		}
+		if err := d.Set("manual_private_link_service_connection", flattenArmPrivateLinkEndpointServiceConnection(privateEndpointProperties.ManualPrivateLinkServiceConnections)); err != nil {
+			return fmt.Errorf("Error setting `manual_private_link_service_connection`: %+v", err)
+		}
+		if err := d.Set("network_interface_ids", flattenArmPrivateLinkEndpointInterface(privateEndpointProperties.NetworkInterfaces)); err != nil {
+			return fmt.Errorf("Error setting `network_interface_ids`: %+v", err)
+		}
+		if subnet := privateEndpointProperties.Subnet; subnet != nil {
+			d.Set("subnet_id", subnet.ID)
+		}
+	}
+	tags.FlattenAndSet(d, resp.Tags)
+
+	return nil
+}
+
+func resourceArmPrivateLinkEndpointDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).Network.PrivateEndpointClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["privateEndpoints"]
+
+	future, err := client.Delete(ctx, resourceGroup, name)
+	if err != nil {
+		if response.WasNotFound(future.Response()) {
+			return nil
+		}
+		return fmt.Errorf("Error deleting Private Endpoint %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		if !response.WasNotFound(future.Response()) {
+			return fmt.Errorf("Error waiting for deleting Private Endpoint %q (Resource Group %q): %+v", name, resourceGroup, err)
+		}
+	}
+
+	return nil
+}
+
+func expandArmPrivateLinkEndpointServiceConnection(input []interface{}) *[]network.PrivateLinkServiceConnection {
+	results := make([]network.PrivateLinkServiceConnection, 0)
+	for _, item := range input {
+		v := item.(map[string]interface{})
+		privateLinkServiceID := v["private_link_service_id"].(string)
+		groupIds := v["group_ids"].([]interface{})
+		requestMessage := v["request_message"].(string)
+		name := v["name"].(string)
+
+		result := network.PrivateLinkServiceConnection{
+			Name: utils.String(name),
+			PrivateLinkServiceConnectionProperties: &network.PrivateLinkServiceConnectionProperties{
+				GroupIds:             utils.ExpandStringSlice(groupIds),
+				PrivateLinkServiceID: utils.String(privateLinkServiceID),
+				RequestMessage:       utils.String(requestMessage),
+			},
+		}
+
+		results = append(results, result)
+	}
+	return &results
+}
+
+func flattenArmPrivateLinkEndpointServiceConnection(input *[]network.PrivateLinkServiceConnection) []interface{} {
+	results := make([]interface{}, 0)
+	if input == nil {
+		return results
+	}
+
+	for _, item := range *input {
+		v := make(map[string]interface{})
+
+		if name := item.Name; name != nil {
+			v["name"] = *name
+		}
+		if props := item.PrivateLinkServiceConnectionProperties; props != nil {
+			if groupIds := props.GroupIds; groupIds != nil {
+				v["group_ids"] = utils.FlattenStringSlice(groupIds)
+			}
```

Comment:
not a blocker but i do prefer to use v
```
			if v := props.GroupIds; v != nil {
				v["group_ids"] = utils.FlattenStringSlice(v)
			}
```

as its easier to review

Reply:
Fixed.